### PR TITLE
docs: formalize release process

### PR DESCRIPTION
**Pull request by dino-rodriguez**
## High Level Overview of Change

We were only documenting the technical part of the release process in `release.md`. 

Added comms & deploy to finish formalizing this process.

### Context of Change

To nail down our release process for PayID.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [x] Documentation Updates
- [ ] Release

## Before / After

Added comms & deploy steps to release process

### CODE REVIEW

File: release.md [hunk 1]
```diff
@@ -24,18 +10,58 @@ The full process for cutting a release is as follows:
...
+
 6. Push up the tag from `master`:
    `git push origin v1.4`
+
 7. Cut a release draft:
    `npm run release`
+
 8. Fill in the release details and publish it in GitHub.
+
+## Deploying the PayID Server
```
Comment (hbergren) [inline release.md:36]:
```suggestion
## Deploying the Xpring PayID Server (Xpring Employees)
```

This isn't relevant to the reference implementation as an open source project, so I would just indicate that.

File: release.md [hunk 2]
```diff
@@ -24,18 +10,58 @@ The full process for cutting a release is as follows:
...
+   as the relevant tag ( e.g. `v1.3` ). The click the `Run Pipeline` button.
+
+5. This should automatically kick off a pipeline that deploys `dev`. From within the pipeline, click the `publish stage image` button, and after that is complete click the `deploy to stage` button.
+
+6. If all goes well, repeat the same for the `prod` buttons in the pipeline.
+
+## Communicating a Release
+
+1. Advertise to internal stakeholders. Refer to this [document](https://docs.google.com/document/d/1QjGc3VBqDARXVbTn0RcQZ164Vu5xNxaWKtmv6ouF8Mo/edit#) for details.
+
```
Comment (hbergren) [inline release.md:54]: Add something about posting on the public Discord? (Not just Ripple-internal Slack channels?)

File: release.md (hunk 1 above)
Comment (dino-rodriguez) [reply release.md:36]: Good point, committing this

File: release.md (hunk 2 above)
Comment (dino-rodriguez) [reply release.md:54]: That should be mentioned in the linked document, but I can make it explicit here as well

**Pull request merged**